### DISCUSSION TITLE
How to uninstall a mod?

### DISCUSSION THREAD

**Discussion by stc1988** [Moddable-OpenSource/moddable#846] [Q&A] [answered]
mcrun can install a mod to change existing behavior without flashing device firmware, on the other hand, how do I revert to before installing a mod?

**Comment by phoddie** [comment, marked as answer]
You can uninstall the current mod using `serial2xsbug` directly. For an ESP32, that looks like this:

```
serial2xsbug /dev/cu.usbserial-0001 460800 8N1 -uninstall
```

Please remember to update the serial path to match your system configuration.

**Comment by stc1988** [comment]
Thank your for replay.
I confirmed that I can uninstall mod using `serial2xsbug`.

I have additional questions.
* Is there any document about `serial2xsbug` ?
* Is there a easier way to detect serial path?

**Comment by phoddie** [comment]
> Thank your for replay.
> I confirmed that I can uninstall mod using serial2xsbug.

Great, thank you.

> Is there any document about serial2xsbug ?

No, not at this time.

> Is there a easier way to detect serial path?

`esptool.py chip_id` works well. That approach may find its way into [xs-dev](https://github.com/HipsterBrown/xs-dev/issues/12#issuecomment-1030685691).
